src/core/ext/client_config/client_channel.c

```diff
@@ -624,18 +658,29 @@ static bool pick_subchannel(grpc_exec_ctx *exec_ctx, grpc_call_element *elem,
   GPR_ASSERT(error == GRPC_ERROR_NONE);
   if (chand->lb_policy != NULL) {
     grpc_lb_policy *lb_policy = chand->lb_policy;
-    int r;
     GRPC_LB_POLICY_REF(lb_policy, "pick_subchannel");
     gpr_mu_unlock(&chand->mu);
+    // If the application explicitly set wait_for_ready, use that.
+    // Otherwise, if the service config specified a value for this
+    // method, use that.
+    if ((initial_metadata_flags &
```

Comment:
I find the body of the `if` difficult to parse. The comment above helps, but I suggest extracting the components of the `if` into separate boolean variables. E.g.

```
const bool wait_for_ready_explicitly_set = initial_metadata_flags & GRPC_INITIAL_METADATA_WAIT_FOR_READY_EXPLICITLY_SET;
const bool call_wait_for_ready_set  = calld->wait_for_ready_from_service_config != WAIT_FOR_READY_UNSET;

if (wait_for_ready_explicitly_set && call_wait_for_ready_set) { ... }
```
... with hopefully better bool variable names :)